app/scripts/controllers/TabSubject.js

```diff
@@ -11,13 +13,64 @@ angular.module('admissionSystemApp')
         templateUrl: '../views/modal/TabSubject.html',
         controller: 'ModalSubjectCtrl',
         size: size,
-        scope: $scope
-      });
+        scope: $scope.$new(true)
+      }).result.then(function (item) {
+          var arrayForId = [],
+            i;
+
+          if (item.allSubjects.addName) {
+            arrayForId = [item.allSubjects.subject].concat(item.allSubjects.addName);
+          }
+          else {
+            arrayForId = [item.allSubjects.subject];
+          }
+
+          for (i = 0; i < arrayForId.length; i++) {  //forEach
+            $scope.entireSpecoffer.subjects.push({
+              enrolmentSubjectId: arrayForId[i].id,
+              // addName: $scope.allSubjects.addName.id,
+              mark: item.mark,
+              isMajor: item.isMajor || false,
+              alternative: item.alternative || false,
+              note: '',
+              weightSubject: item.weightSubject
+            });
+
+            console.log('id' + $scope.entireSpecoffer.subjects[i].enrolmentSubjectId);
+
+            showSubjectById($scope.entireSpecoffer.subjects[$scope.entireSpecoffer.subjects.length - 1].enrolmentSubjectId,
```

Comment:
You have to extract $scope.entireSpecoffer.subjects[$scope.entireSpecoffer.subjects.length - 1] to variable